src/components/TextInput/index.tsx

```diff
@@ -0,0 +1,84 @@
+import React, {Component, ForwardedRef, useEffect, useRef} from 'react';
+import {StyleProp, TextInputProps, ViewStyle} from 'react-native';
+import {AnimatedProps} from 'react-native-reanimated';
+import * as Browser from '@libs/Browser';
+import DomUtils from '@libs/DomUtils';
+import Visibility from '@libs/Visibility';
+import useThemeStyles from '@styles/useThemeStyles';
+import BaseTextInput from './BaseTextInput';
+import BaseTextInputProps from './BaseTextInput/types';
+import * as styleConst from './styleConst';
+
+function TextInput(
+    {label = '', name = '', textInputContainerStyles, inputStyle, disableKeyboard = false, multiline = false, prefixCharacter, inputID, ...props}: BaseTextInputProps,
+    ref: ForwardedRef<HTMLFormElement | Component<AnimatedProps<TextInputProps>, unknown, unknown>>,
+) {
+    const styles = useThemeStyles();
+    const textInputRef = useRef<HTMLFormElement>(null);
+    const removeVisibilityListenerRef = useRef<() => void>(null);
+
+    useEffect(() => {
+        if (disableKeyboard) {
+            textInputRef.current?.setAttribute('inputmode', 'none');
+        }
+
+        if (name) {
+            textInputRef.current?.setAttribute('name', name);
+        }
+        // @ts-expect-error We need to reassign this ref to the input ref
+        removeVisibilityListenerRef.current = Visibility.onVisibilityChange(() => {
```

Comment:
This way you can get rid of @ts-expect-error
```
type RemoveVisibilityListener = () => void;
....
const removeVisibilityListenerRef = useRef<RemoveVisibilityListener>(null);
...
(removeVisibilityListenerRef.current as RemoveVisibilityListener)  = Visibility.onVisibilityChange(() => {

Reply:
@VickyStash I fixed that using different approach since what you suggested was causing another error. Let me know what you think 😄